src/backend/distributed/sql/udfs/citus_job_status/latest.sql

```diff
@@ -0,0 +1,98 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_job_status (
+    job_id bigint,
+    raw boolean DEFAULT FALSE
+)
+    RETURNS TABLE (
+            job_id bigint,
+            state pg_catalog.citus_job_status,
+            job_type name,
+            description text,
+            started_at timestamptz,
+            finished_at timestamptz,
+            details jsonb
+    )
+    LANGUAGE SQL
+    STRICT
+    AS $fn$
+    WITH rp AS MATERIALIZED (
+        SELECT
+            sessionid,
+            sum(source_shard_size) as source_shard_size,
+            sum(target_shard_size) as target_shard_size,
+            any_value(status) as status,
+            any_value(sourcename) as sourcename,
+            any_value(sourceport) as sourceport,
+            any_value(targetname) as targetname,
+            any_value(targetport) as targetport,
+            max(source_lsn) as source_lsn,
+            min(target_lsn) as target_lsn
+        FROM get_rebalance_progress()
+        GROUP BY sessionid
+    ),
+    task_state_occurence_counts AS (
+        SELECT t.status, count(task_id)
+        FROM pg_dist_background_job j
+            JOIN pg_dist_background_task t ON t.job_id = j.job_id
+        WHERE j.job_id = $1
+        GROUP BY t.status
+    ),
+    running_task_details AS (
+        SELECT jsonb_agg(jsonb_build_object(
+                'state', t.status,
+                'retried', coalesce(t.retry_count,0),
+                'phase', rp.status,
+                'size' , jsonb_build_object(
+                    'source', rp.source_shard_size,
+                    'target', rp.target_shard_size),
+                'hosts', jsonb_build_object(
+                    'source', rp.sourcename || ':' || rp.sourceport,
+                    'target', rp.targetname || ':' || rp.targetport),
+                'message', t.message,
+                'command', t.command,
+                'LSN', jsonb_build_object(
+                    'source', rp.source_lsn,
+                    'target', rp.target_lsn),
```

Comment:
target_lsn is `0/0` in many cases, I think a better value would be NULL. But this should probably be fixed directly in get_rebalance_progress()

@agedemenli what do you think?

Reply:
makes sense
we can do a simple `if ( targetLSN == InvalidXLogRecPtr)` in get_rebalance_progress

Reply:
@agedemenli will create a separate PR for this.

Reply:
merged #6617